src/components/Menus/Sidebar/InstructionsPanel/InstructionsPanel.jsx

```diff
@@ -1,30 +1,36 @@
 import React, { useEffect, useRef, useMemo, useState } from "react";
 import SidebarPanel from "../SidebarPanel";
 import { useTranslation } from "react-i18next";
-import { useSelector } from "react-redux";
+import { useSelector, useDispatch } from "react-redux";
 import ProgressBar from "./ProgressBar/ProgressBar";
 import "../../../../assets/stylesheets/Instructions.scss";
 import "prismjs/plugins/highlight-keywords/prism-highlight-keywords.js";
 import "prismjs/plugins/line-highlight/prism-line-highlight";
 import "prismjs/plugins/line-numbers/prism-line-numbers.css";
 import "prismjs/plugins/line-highlight/prism-line-highlight.css";
 import { quizReadyEvent } from "../../../../events/WebComponentCustomEvents";
+import { setCurrentStepPosition } from "../../../../redux/InstructionsSlice";
 
 const InstructionsPanel = () => {
   const steps = useSelector((state) => state.instructions.project?.steps);
   const quiz = useSelector((state) => state.instructions?.quiz);
-
+  const dispatch = useDispatch();
   const currentStepPosition = useSelector(
     (state) => state.instructions.currentStepPosition,
   );
   const { t } = useTranslation();
   const stepContent = useRef();
-  const [quizReady, setQuizReady] = useState(false);
 
-  const isQuiz = useMemo(() => {
-    return !!quiz?.questionCount;
+  const [isQuiz, setIsQuiz] = useState(false);
+
+  const quizCompleted = useMemo(() => {
+    return quiz.currentQuestion === quiz?.questionCount;
```

Comment:
Does the first half of this need a guard against `quiz` being blank as well?

Reply:
I've added optional chaining to the first half.